scheduler/host_allocator_fuzzer_test.go

```diff
@@ -0,0 +1,157 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/distro"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	taskDurations = []time.Duration{
+		10 * time.Second,
+		1 * time.Minute,
+		5 * time.Minute,
+		10 * time.Minute,
+		15 * time.Minute,
+		20 * time.Minute,
+		30 * time.Minute,
+		1 * time.Hour,
+	}
+	expectedDurationMax       = 1 * time.Hour
+	maxNumTasks               = 100 // max # of tasks in the queue
+	maxRunningHosts           = 100 // max # of hosts running a task
+	maxStartTimeOffsetMinutes = 30  // max time before now that a task has been running
+	maxNumFreeHosts           = 5   // max # of hosts running no task
+	numIterations             = 100 // # of tests to run
```

Comment:
shouldn't these be set in the constructor for the suite so that we could run it with different settings?

Reply:
changed